# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: 1f21e8f610a7
# Full Hash: 1f21e8f610a72d6c6bd84321f41fb257335213e4
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-31 21:05:17
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r c50443ca5619 -r 1f21e8f610a7 js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Sat Oct 31 09:56:05 2020 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Sat Oct 31 09:56:05 2020 +0000
@@ -65,7 +65,7 @@
   const CodeRange* codeRange_;
   unsigned lineOrBytecode_;
   Frame* fp_;
-  TlsData* tls_;
+  const TlsData* tls_;
   uint8_t* unwoundIonCallerFP_;
   jit::FrameType unwoundIonFrameType_;
   Unwind unwind_;
@@ -96,7 +96,7 @@
   jit::FrameType unwoundIonFrameType() const;
   uint8_t* unwoundIonCallerFP() const { return unwoundIonCallerFP_; }
   Frame* frame() const { return fp_; }
-  TlsData* tls() const { return tls_; }
+  const TlsData* tls() const { return tls_; }
 
   // Returns the address of the next instruction that will execute in this
   // frame, once control returns to this frame.
@@ -237,6 +237,7 @@
 
 // Iterates through frames for either possible cross-instance call or an entry
 // stub to obtain tls that corresponds to the passed fp.
+const TlsData* GetNearestEffectiveTls(const Frame* fp);
 TlsData* GetNearestEffectiveTls(Frame* fp);
 
 // Describes register state and associated code at a given call frame.